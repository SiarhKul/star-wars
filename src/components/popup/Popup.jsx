import React from "react";
import { useDispatch } from "react-redux";
import {
	isPopupVisibleOutsideClick,
	isVisiblePopup,
} from "../../redux/actionsCreators/actionsCrators";

export const Popup = ({ FragmentPopup }) => {
	const dispatch = useDispatch();

	return (
		<div
			className="popup"
			data-popup="popup"
			onClick={e => dispatch(isPopupVisibleOutsideClick(e))}
		>
			<div className="popup-window">
				<div
					className="exit"
					onClick={() => {
						dispatch(isVisiblePopup(false));
					}}
				>
					&#10060;
				</div>

				<FragmentPopup />
			</div>
		</div>
	);
};
