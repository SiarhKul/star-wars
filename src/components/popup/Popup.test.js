import { shallow } from "enzyme";
import React from "react";
import { Popup } from "./Popup";

jest.mock("react-router", () => ({
	...jest.requireActual("react-router"),
	useParams: jest.fn().mockReturnValue({ id: 1 }),
}));

describe("Test <Popup/>", () => {
	const mockedProps = { FragmentPopup: jest.fn() };
	const component = shallow(<Popup {...mockedProps} />);

	it("render snapshot <Popup/>", () => {
		expect(component).toMatchSnapshot();
	});

	it("should hide <Popup/>  on click if event is no undefended ", () => {
		const e = { target: { dataset: { popup: "popup" } } };
		const isPopupVisibleOutSideClickMemo = component.prop("onClick");
		const mockFn = jest.fn(isPopupVisibleOutSideClickMemo);
		expect(mockFn.mock.calls.length).toBe(0);
		mockFn(e);
		expect(mockFn.mock.calls.length).toBe(1);
	});

	it("should hide <Popup/>  on click if event is undefended ", () => {
		const e = { target: { dataset: { popup: "" } } };
		const isPopupVisibleOutSideClickMemo = component.prop("onClick");
		const mockFn = jest.fn(isPopupVisibleOutSideClickMemo);
		expect(mockFn.mock.calls.length).toBe(0);
		mockFn(e);
		expect(mockFn.mock.calls.length).toBe(1);
	});

	it("should hide <Popup/> on click close button ", () => {
		const closeButton = component.find(".exit__btn");
		const returnToRootPath = closeButton.prop("onClick");
		const mockFn = jest.fn(returnToRootPath);
		expect(mockFn.mock.calls.length).toBe(0);
		mockFn();
		expect(mockFn.mock.calls.length).toBe(1);
	});
});
