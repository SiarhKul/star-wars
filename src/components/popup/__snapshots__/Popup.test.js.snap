// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Popup/> render snapshot <Popup/> 1`] = `
<div
  className="popup"
  data-popup="popup"
  onClick={[Function]}
>
  <div
    className="popup-window"
  >
    <div
      className="exit"
      onClick={[Function]}
    >
      <button
        className="exit__btn"
      >
         ‚ùå
      </button>
    </div>
    <mockConstructor />
  </div>
</div>
`;
